### PR TITLE
Opening Hours - Add AM/PM format

### PR DESCRIPTION
**Pull request by Skalii**

### CODE REVIEW

File: Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm [hunk 1]
```diff
@@ -157,19 +158,42 @@ - (NSAttributedString *) getAdditionalInfoStr
         NSMutableAttributedString *str = [[NSMutableAttributedString alloc] init];
         UIColor *colorOpen = UIColorFromRGB(color_ctx_menu_amenity_opened_text);
         UIColor *colorClosed = UIColorFromRGB(color_ctx_menu_amenity_closed_text);
-        for (auto info : _openingHoursInfo)
+        for (int i = 0; i < _openingHoursInfo.size(); i ++)
         {
+            auto info = _openingHoursInfo[i];
+
             if (str.length > 0)
```
Comment (PaulStets) [inline Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm:165]: This should be done using NSDateFormatter (see https://developer.apple.com/documentation/foundation/nsdateformatter/1409506-amsymbol)

File: Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm [hunk 2]
```diff
@@ -157,19 +158,42 @@ - (NSAttributedString *) getAdditionalInfoStr
...
+            NSMutableAttributedString *s = [[NSMutableAttributedString alloc] initWithString:[NSString stringWithFormat:@"  %@", time]];
             NSTextAttachment *attachment = [[NSTextAttachment alloc] init];
-            attachment.image = [OAUtilities tintImageWithColor:[UIImage imageNamed:@"ic_travel_time"] color:info->opened ? colorOpen : colorClosed];
+            BOOL opened = info->fallback && i > 0 ? _openingHoursInfo[i - 1]->opened : info->opened;
+            attachment.image = [OAUtilities tintImageWithColor:[UIImage imageNamed:@"ic_travel_time"]
+                                                                             color:opened ? colorOpen : colorClosed];
             
             NSAttributedString *strWithImage = [NSAttributedString attributedStringWithAttachment:attachment];
             [s replaceCharactersInRange:NSMakeRange(0, 1) withAttributedString:strWithImage];
-            [s addAttribute:NSBaselineOffsetAttributeName value:[NSNumber numberWithFloat:-2.0] range:NSMakeRange(0, 1)];
```
Comment (PaulStets) [inline Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm:171]: Do not do this, please. We have discussed pointless code changes multiple times!

File: Sources/Helpers/OAExternalTimeFormatter.mm [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+{
+    std::vector<std::string> months;
+    for (int i = 0; i < 12; i++)
+    {
+        months.push_back([self.class getTranslatesMonth:i].UTF8String);
+    }
+    return months;
+}
+
++ (NSString *) getTranslatesWeekday:(int)number
```
Comment (alex-osm) [inline Sources/Helpers/OAExternalTimeFormatter.mm:79]: Return NSArray<NSString *>

File: Sources/Helpers/OAExternalTimeFormatter.mm [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    NSCalendar *calendar = [NSCalendar currentCalendar];
+    NSDateComponents *components = [calendar components:(NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear) fromDate:date];
+    components.weekday = number;
+    date = [calendar dateFromComponents:components];
+    NSDateFormatter *formatter = [[NSDateFormatter alloc]init];
+    [formatter setDateFormat:@"EEE"];
+    return [formatter stringFromDate:date];
+}
+
++ (NSString *) getTranslatesMonth:(int)number
```
Comment (alex-osm) [inline Sources/Helpers/OAExternalTimeFormatter.mm:91]: Return NSArray<NSString *>

**Pull request merged**